### DISCUSSION TITLE
fluidsynth how to process multi midi at the same time?

### DISCUSSION THREAD

**Discussion by nijian141893** [FluidSynth/fluidsynth#1099] [Q&A] [answered]
for example

  ```
 fluid_synth_noteon(synth, 0, 60, 127); // play middle C
 sleep(1); // sleep for 1 second
 fluid_synth_noteoff(synth, 0, 60); // stop playing middle C
```

this code can process a midi input to a note, which we can hear that.

but, If I input several midi event, and I want play them at the same time.

so,  how could I do this, thks.

**Comment by derselbst** [comment, marked as answer]
Doing this "at the same time" is only possible when dispatching events from within the [synth context](https://www.fluidsynth.org/api/synth-context.html).

Since your use-case comprises an audio driver which continuously renders audio, and another thread which continuously delivers events, the events may or may not be dispatched at the same time. If I were you, I wouldn't bother about this, because that minor time difference will probably be inaudible.
